model/task/db.go

```diff
@@ -1820,31 +1820,41 @@ func FindProjectForTask(taskID string) (string, error) {
 	return t.Project, nil
 }
 
-func updateAllMatchingDependenciesForTask(taskId, dependencyId string, unattainable bool) error {
+func (t *Task) updateAllMatchingDependenciesForTask(dependencyId string, unattainable bool) error {
```

Comment:
i see that tests cover MarkUnattainableDependency; is there any case we'd want to introduce for that or for this function that might catch more of what we're trying to fix here? (i.e. that would fail with the old function)

Reply:
done